src/session/content/message_row/sticker.rs

```diff
@@ -109,69 +124,44 @@ impl MessageBaseExt for MessageSticker {
             // FIXME: Do not show message reply when message is being deleted
             // Sticker and the reply should be at the opposite sides of the box
             if message.is_outgoing() {
-                reply.insert_before(self, Some(&imp.sticker_overlay.get()));
+                reply.insert_before(self, Some(&imp.overlay.get()));
             } else {
-                reply.insert_after(self, Some(&imp.sticker_overlay.get()));
+                reply.insert_after(self, Some(&imp.overlay.get()));
             }
         }
-        imp.picture.set_texture(None);
-
-        if let MessageContent::MessageSticker(data) = message.content().0 {
-            self.imp()
-                .picture
-                .set_aspect_ratio(data.sticker.width as f64 / data.sticker.height as f64);
 
-            if data.sticker.sticker.local.is_downloading_completed {
-                self.load_sticker(data.sticker.sticker.local.path);
-            } else {
-                let file_id = data.sticker.sticker.id;
-                let session = message.chat().session();
-
-                spawn(clone!(@weak self as obj, @weak session => async move {
-                    obj.download_sticker(file_id, &session).await;
-                }));
+        let (sticker, looped, is_emoji) = match message.content().0 {
+            MessageContent::MessageSticker(data) => {
+                let sticker = data.sticker;
+                (sticker, true, false)
             }
-        }
-
-        self.notify("message");
-    }
-}
-
-impl MessageSticker {
-    async fn download_sticker(&self, file_id: i32, session: &Session) {
-        match session.download_file(file_id).await {
-            Ok(file) => {
-                self.load_sticker(file.local.path);
+            MessageContent::MessageAnimatedEmoji(data) => {
+                let sticker = data.animated_emoji.sticker.unwrap();
+                let looped = matches!(sticker.full_type, StickerFullType::CustomEmoji(_));
+                (sticker, looped, true)
             }
-            Err(e) => {
-                log::warn!("Failed to download a sticker: {e:?}");
+            _ => unreachable!(),
+        };
+
+        match &sticker.full_type {
+            StickerFullType::CustomEmoji(data) if data.needs_repainting => {
+                self.add_css_class("needs-repainting")
             }
+            _ => self.remove_css_class("needs-repainting"),
         }
```

Comment:
Mh, I don't really get what this css class is for.

Reply:
Check one of those packs: Emoticon Emoji, Outline Emoji, Animal Emoji

Those emojis can be either black or white dependent on theme, currently I'm using css to recolor them

Reply:
[Screencast from 2023-03-24 09-14-40.webm](https://user-images.githubusercontent.com/63008755/227440427-ae956233-40b4-4db6-82bf-fc9dfba344b2.webm)